Upstreaming: Copy common codebase

3mkusiak opened the pull request:
Hard-copy common codebase from RPI demos to NXP ones.
Fix `readme`.

3mkusiak committed:
Copy common codebase

This commit mainly adds hard-copied common codebase between RPI and NXP
solutions. Additionally readme was updated to remove "FIXMEs" and link
to pubilc repository has been updated.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

3mkusiak commented:
Prove the uc1-1 works: 
[uc1-1_repro.zip](https://github.com/user-attachments/files/22720294/uc1-1_repro.zip)

3mkusiak commented:
The `tls_client/src/common` does not have to be reviewed. It has been copied 1:1 from other repo.

m-iwanicki commented on README.md:
```diff
@@ -61,19 +61,11 @@ This section instructs on how to build the final demo for Use Case 1.1. As a
-
-Make sure the common codebase has been synced, a `tls_client/src/common` should
-be created and have contents.
+[can be found here](https://github.com/crosscon/uc1-2-integration).
```
Mention that you have to build `NXP_PUF` demo

m-iwanicki reviewed (commented):
@3mkusiak I get build errors:

```sh
export TARGET=246; ./uc1.sh build_mtls_puf
```

Output:

```
(...)
[264/264] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       40996 B       160 KB     25.02%
             RAM:       19928 B        64 KB     30.41%
        USB_SRAM:          0 GB        16 KB      0.00%
        IDT_LIST:          0 GB        32 KB      0.00%
Generating files from /home/miwanicki/projects/crosscon/workspace/build/zephyr/zephyr.elf for board: lpcxpresso55s69
image.py: sign the payload
image.py: sign the payload
VM1_START extracted from zephyr.elf: 0x000225f4
Check placeholder!
*---- ACTIVATION CODE ----*
Unable to find image 'bao-hypervisor-image:latest' locally
docker: Error response from daemon: pull access denied for bao-hypervisor-image, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
Failed to run 'arm-none-eabi-objdump -s -j .activation_code /home/miwanicki/projects/crosscon/workspace/uc1-integration/out/vm1.elf' in Docker image: bao-hypervisor-image : (125)
```

3mkusiak committed:
README: Add info on building container image

The information on how to build the container image is burried deep into
the guides. Add information on how to do so in readme.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

m-iwanicki reviewed (changes requested):
One comment to address. I managed to build everything.

3mkusiak committed:
README: name demo explicitly

The users must know which demo they need to build for RPI platform, add
the demo name explicitly.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>
